esmf_utils/ESMF_Time_Utilities.F90

```diff
@@ -0,0 +1,96 @@
+#include "MAPL_Generic.h"
+module mapl3g_ESMF_Time_Utilities
+   use esmf, I4 => ESMF_KIND_I4
+   use mapl_ErrorHandling
+   implicit none (type, external)
+!   private !wdb fixme deleteme should this be private
+
+   public :: zero_time_interval
+   public :: intervals_are_compatible
+   public :: times_and_intervals_are_compatible
+
+   interface zero_time_interval
+      module procedure :: get_zero
+   end interface zero_time_interval
+
+   integer, parameter :: NUM_DATETIME_FIELDS = 6
+
+   type :: DateTimeCheck
+   end type DateTimeCheck
+
+   ! This value should not be accessed directly. Use get_zero() instead.
+   ! There is no constructor for ESMF_TimeInterval, so the value cannot be initialized
+   ! at construction. The get_zero() function initializes the value the first time
+   ! and returns a pointer to the value.
+   type(ESMF_TimeInterval), target :: ZERO_TI
+
+contains
+
+   logical function can_compare_intervals(larger, smaller, rc)
+      type(ESMF_TimeInterval), intent(inout) :: larger
+      type(ESMF_TimeInterval), intent(inout) :: smaller
+      integer, optional, intent(out) :: rc
+      integer :: status
+      integer(kind=I4) :: lyy, lmm, ld, lh, lm, ls
+      integer(kind=I4) :: syy, smm, sd, sh, sm, ss
```

Comment:
Presumably one of these was meant to be I8?

I really do think you would benefit from compiling as you go ...

Reply:
It compiles and passes the MAPL tests on bucy for ifort, nagfor, and gfortran. The tests on GitHub are still running, and sometimes, they don't pass even if they pass on bucy. It may be that on GitHub it won't compiler. Do we want to use I4 or I8 for getting and setting TimeInterval and Time? Our current code seems to use I4 for datetime units, but it is always safer to use I8.